psycop/common/model_training_v2/trainer/task/estimator_steps/xgboost.py

```diff
@@ -1,16 +1,29 @@
-from typing import Any
+from typing import Literal
 
 import numpy as np
 from xgboost import XGBClassifier
 
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
 from psycop.common.model_training_v2.trainer.task.model_step import (
     ModelStep,
 )
 
 
-def xgboost_classifier_step(**kwargs: Any) -> ModelStep:
+@BaselineRegistry.estimator_steps.register("xgboost")
+def xgboost_classifier_step(
+    tree_method: Literal["auto", "gpu_hist"] = "gpu_hist",
+    n_estimators: int = 100,
+    max_depth: int = 3,
+) -> ModelStep:
     """Initialize XGBClassifier model with hparams specified as kwargs.
-    The 'missing' hyperparameter specifies the value to be treated as missing
-    and is set to np.nan by default."""
-    static_hyperparameters: dict[str, float] = {"missing": np.nan}
-    return ("xgboost", XGBClassifier(**kwargs, **static_hyperparameters))
+    The 'missing' hyperparameter specifies the value to be treated as missing and is set to np.nan by default.
```

Comment:
meaning only the values corresponding to the "missing" value are marked as missing?

Reply:
Yep!